src/linker/Linker/Driver.cs

```diff
@@ -780,29 +780,37 @@ public int Run (ILogger? customLogger = null, bool throwOnFatalLinkerException =
 
 			try {
 				p.Process (Context);
-			} catch (LinkerFatalErrorException lex) {
+			} catch (Exception e) when (LogFatalError (e)) {
+				// Unreachable
+				throw;
+			}
+
+			Context.FlushCachedWarnings ();
+			Context.Tracer.Finish ();
+			return Context.ErrorsCount > 0 ? 1 : 0;
+		}
+
+		/// <summary>
+		/// This method is called in the exception filter for unexpected exceptions.
+		/// Prints error messages and returns false to avoid catching in the exception filter.
+		/// </summary>
+		bool LogFatalError (Exception e)
+		{
+			switch (e) {
+			case LinkerFatalErrorException lex:
 				Context.LogMessage (lex.MessageContainer);
-				Console.Error.WriteLine (lex.ToString ());
```

Comment:
Why delete this line? Is it just a duplicate for some reason?

Reply:
My understanding is that when the exception causes the program to crash, it will print out all the same information as lex.ToString(), so it isn't necessary to do it in the exception filter too.